gatherers/manifest.js

```diff
@@ -54,34 +32,62 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
```

Comment:
This feels like we should be using `Promise.race` here. As it stands this feels very side-effecty, especially as it returns both the listener and the promise.